virttest/vt_resmgr/resources/storage/file_volume.py

```
@@ -0,0 +1,231 @@
+    def allocate(self, arguments, node):
+        # Pass size as an argument to its backing, because size can change
+        arguments = arguments if arguments else dict()
+        arguments["size"] = self.spec["size"]
+        node, backing_id = self.get_binding(node)
+        r, o = node.proxy.resource.update_resource_by_backing(
+            backing_id,
+            "allocate",
+            arguments,
+        )
+        if r != 0:
+            raise Exception(o["out"])
```

Ruff (0.13.1)

Create your own exception

(TRY002)